specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimdeployment.json

```diff
@@ -0,0 +1,1400 @@
+        "x-ms-examples": {
+          "ApiManagementServiceUpdateHostName": {
+            "$ref": "./examples/ApiManagementServiceUpdateHostName.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApiManagementServiceUpdateHostnameParameters"
+            },
+            "description": "Parameters supplied to the UpdateHostname operation."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Service was successfully updated with desired hostnames.",
+            "schema": {
+              "$ref": "#/definitions/ApiManagementServiceResource"
+            }
+          },
+          "202": {
+            "description": "Accepted. The location header contains the URL where the status of the long running operation can be checked. The response also includes the unmodified ApiManagementServiceResource.",
+            "schema": {
+              "$ref": "#/definitions/ApiManagementServiceResource"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "CertificateInformation": {
+      "properties": {
+        "expiry": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Expiration date of the certificate. The date conforms to the following format: `yyyy-MM-ddTHH:mm:ssZ` as specified by the ISO 8601 standard."
+        },
+        "thumbprint": {
+          "type": "string",
+          "description": "Thumbprint of the certificate."
+        },
+        "subject": {
+          "type": "string",
+          "description": "Subject of the certificate."
+        }
+      },
+      "required": [
+        "expiry",
+        "thumbprint",
+        "subject"
+      ],
+      "description": "SSL certificate information."
+    },
+    "CertificateConfiguration": {
+      "properties": {
+        "encodedCertificate": {
+          "type": "string",
+          "description": "Base64 Encoded certificate."
+        },
+        "certificatePassword": {
+          "type": "string",
+          "description": "Certificate Password."
+        },
+        "storeName": {
```

Comment:
>storeName [](start = 9, length = 9)

Should x-ms-enum be used?

Reply:
done

Reply:
I mean using this extension
[https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum)


---
In reply to: [168378244](https://github.com/Azure/azure-rest-api-specs/pull/2292#discussion_r168378244) [](ancestors = 168378244)

Reply:
what should be the name of the `x-ms-enum` parameter you want us to specify. If it is `StoreName` then that is already well known object in .NET. Also if the customer uses it in the same file, he will have to do explicit namereferencing like

Microsoft.WindowsAzure.ApiManagement.Modes.StoreName 

or System.Cryptography......StoreName and we dont want that experience for the customer.



---
In reply to: [168617424](https://github.com/Azure/azure-rest-api-specs/pull/2292#discussion_r168617424) [](ancestors = 168617424,168378244)